WordPress/Classes/Networking/MenusServiceRemote.m

```diff
@@ -0,0 +1,379 @@
+#import "MenusServiceRemote.h"
+#import "Blog.h"
+#import "WordPressComApi.h"
+#import "RemoteMenu.h"
+#import "RemoteMenuItem.h"
+#import "RemoteMenuLocation.h"
+
+NSString * const MenusRemoteKeyID = @"id";
+NSString * const MenusRemoteKeyMenu = @"menu";
+NSString * const MenusRemoteKeyMenus = @"menus";
+NSString * const MenusRemoteKeyLocations = @"locations";
+NSString * const MenusRemoteKeyContentID = @"content_id";
+NSString * const MenusRemoteKeyDescription = @"description";
+NSString * const MenusRemoteKeyLinkTarget = @"link_target";
+NSString * const MenusRemoteKeyLinkTitle = @"link_title";
+NSString * const MenusRemoteKeyName = @"name";
+NSString * const MenusRemoteKeyType = @"type";
+NSString * const MenusRemoteKeyTypeFamily = @"type_family";
+NSString * const MenusRemoteKeyTypeLabel = @"type_label";
+NSString * const MenusRemoteKeyURL = @"url";
+NSString * const MenusRemoteKeyItems = @"items";
+NSString * const MenusRemoteKeyDeleted = @"deleted";
+NSString * const MenusRemoteKeyLocationDefaultState = @"defaultState";
+
+@implementation MenusServiceRemote
+
+#pragma mark - Remote queries: Creating and modifying menus
+
+- (void)createMenuWithName:(NSString *)menuName
+                      blog:(Blog *)blog
+                   success:(MenusServiceRemoteMenuRequestSuccessBlock)success
+                   failure:(MenusServiceRemoteFailureBlock)failure
+{
+    NSNumber *blogId = [blog dotComID];
+    NSParameterAssert([blogId isKindOfClass:[NSNumber class]]);
+    NSParameterAssert([menuName isKindOfClass:[NSString class]]);
+    
+    NSString *path = [NSString stringWithFormat:@"sites/%@/menus/new", blogId];
+    NSString *requestURL = [self pathForEndpoint:path
+                                     withVersion:ServiceRemoteRESTApiVersion_1_1];
+    
+    [self.api POST:requestURL
+        parameters:@{MenusRemoteKeyName: menuName}
+           success:^(AFHTTPRequestOperation * _Nonnull operation, id  _Nonnull responseObject) {
+               if (success) {
+                   NSAssert([responseObject isKindOfClass:[NSDictionary class]], @"Expected a dictionary");
```

Comment:
Assertions are not great for parsing API responses.  Assertions are mostly useful for internal state verifications, cases in which we can expect to catch the error while testing or unit-testing the app.

The problem with using them for parsing API responses is that the make it possible for server errors to propagate to our app as crashes (meaning a server error could come across to users as a badly-written & crashing iOS app instead of an alert dialog).

PS: I know we're doing this everywhere but I think we should consider switching these assertions to proper `if` blocks.  If the `if` block fails, we could call the failure block with a proper error message.